jinahub/indexers/SimpleIndexer/simple_indexer.py

```diff
@@ -97,6 +97,45 @@ def search(
         )
         self._flush = False
 
+    @requests(on='/delete')
+    def delete(self, docs: DocumentArray, parameters: Optional[Dict] = {}, **kwargs):
+        """Delete entries from the index by id
+
+        :param docs: the documents to delete
+        :param parameters: parameters to the request
+        """
+        if docs is None:
+            return
+        traversal_paths = parameters.get(
+            'traversal_paths', self.default_traversal_paths
+        )
+        flat_docs = docs.traverse_flat(traversal_paths)
+        delete_docs_ids = set(flat_docs.get_attributes('id'))
+        for idx in reversed(range(len(self._docs))):
+            if self._docs[idx].id in delete_docs_ids:
+                del self._docs[idx]
+
+    @requests(on='/update')
+    def update(self, docs: DocumentArray, parameters: Optional[Dict] = {}, **kwargs):
+        """Update doc with the same id
+
+        :param docs: the documents to update
+        :param parameters: parameters to the request
+        """
+        if docs is None:
+            return
+        traversal_paths = parameters.get(
+            'traversal_paths', self.default_traversal_paths
+        )
+        flat_docs = docs.traverse_flat(traversal_paths)
+
+        update_docs_ids = set(flat_docs.get_attributes('id'))
+        for idx in reversed(range(len(self._docs))):
+            if self._docs[idx].id in update_docs_ids:
+                update_doc = flat_docs.get_attributes_with_docs(self._docs[idx].id)[1][0]
+                del self._docs[idx]
+                self._docs.append(update_doc)
```

Comment:
```suggestion
        for doc in flat_docs:
            if doc.id is not None:
                self._docs[doc.id] = doc
            else:
                self._docs.append(doc)
        
```

Reply:
I do agree with @numb3r3 's suggestion.
simple tweak: maybe we should not accept updating docs with `None` ids or with ids that don't exists in the DAM